server/server_test.go

```diff
@@ -2804,6 +2807,48 @@ func validateAuthorizedRequest(t *testing.T, s *Server, req *http.Request, exp i
 	}
 }
 
+func TestServerUsesAuthorizerParsedBody(t *testing.T) {
```

Comment:
It might be good to extend the existing end-to-end-y test we have for authorization? (https://github.com/open-policy-agent/opa/blob/57c5b72948a79b4e8532bc5f97ff934574b0c0a4/server/server_test.go#L2697) WDYT?

Reply:
Extended the E2E test.